CMakeLists.txt

```diff
@@ -221,6 +222,10 @@ if(USE_CUDA)
   add_subdirectory(${PROJECT_SOURCE_DIR}/gputreeshap)
 
   find_package(CUDAToolkit REQUIRED)
+
+  if(BUILD_WITH_EXTERNAL_CCCL)
+    find_package(CCCL REQUIRED CONFIG COMPONENTS cub)
```

Comment:
Would it make sense to always do this (iow dropping `if`)? Think this would pick up the CTK CCCL if no other is found

Reply:
`find_package(CCCL)` errors out if `cccl` isn't installed explicitly, since CMake can't locate `cccl-config.cmake` or `FindCCCL.cmake`.

Reply:
Can we please double check this?

In the Conda provided CTK, we make sure [the CCCL config files are packaged]( https://github.com/conda-forge/cuda-cccl-feedstock/blob/c78dce02651dad7f8f42bcd2156016f747c48b2a/recipe/meta.yaml#L41-L44 )

Would think this is the case in other CTK packages of CCCL. Though even if that were not the case, it would be good to discover and raise with the right folks to fix

Reply:
Yes, when I tried building it locally without installing `cccl`, I got an error about missing `cccl-config.cmake`. I wonder why, especially because the CTK does contain `cccl-config.cmake`:
```
$ find /usr/local/cuda/targets/x86_64-linux/lib/cmake/cccl/
/usr/local/cuda/targets/x86_64-linux/lib/cmake/cccl/
/usr/local/cuda/targets/x86_64-linux/lib/cmake/cccl/cccl-config.cmake
/usr/local/cuda/targets/x86_64-linux/lib/cmake/cccl/cccl-config-version.cmake
```

Reply:
Awesome, thanks Hyunsu! 🙏

Please feel free to mark this thread resolved